src/main/java/techreborn/compatmod/ic2/IC2Dict.java

```diff
@@ -49,90 +43,131 @@
-				ItemParts.getPartByName("rubber", 3), 400, 2));
-
 		try {
-			CraftingItemType.circuit.getName();
+			OreUtil.registerOre("reBattery", getItem("re_battery"));
+
+			OreUtil.registerOre("circuitBasic", getItem("crafting", "circuit"));
+			OreUtil.registerOre("circuitAdvanced", getItem("crafting", "advanced_circuit"));
+
+			OreUtil.registerOre("machineBlockBasic", getItem("resource", "machine"));
+			OreUtil.registerOre("machineBlockAdvanced", getItem("resource", "advanced_machine"));
 
-			OreUtil.registerOre("reBattery", ItemName.re_battery.getItemStack());
+			OreUtil.registerOre("lapotronCrystal", getItem("lapotron_crystal"));
+			OreUtil.registerOre("energyCrystal", getItem("energy_crystal"));
 
-			OreUtil.registerOre("circuitBasic", ItemName.crafting.getItemStack(CraftingItemType.circuit));
-			OreUtil.registerOre("circuitAdvanced", ItemName.crafting.getItemStack(CraftingItemType.advanced_circuit));
+			OreUtil.registerOre("drillBasic", getItem("drill"));
+			OreUtil.registerOre("drillDiamond", getItem("diamond_drill"));
 
-			OreUtil.registerOre("machineBlockBasic", BlockName.resource.getItemStack(ResourceBlock.machine));
-			OreUtil.registerOre("machineBlockAdvanced", BlockName.resource.getItemStack(ResourceBlock.advanced_machine));
+			if(experimental) {
+				OreUtil.registerOre("drillAdvanced", getItem("iridium_drill"));
+			}
 
-			OreUtil.registerOre("lapotronCrystal", ItemName.lapotron_crystal.getItemStack());
-			OreUtil.registerOre("energyCrystal", ItemName.energy_crystal.getItemStack());
+			OreUtil.registerOre("reflectorBasic", getItem("neutron_reflector"));
+			OreUtil.registerOre("reflectorThick", getItem("thick_neutron_reflector"));
 
-			OreUtil.registerOre("drillBasic", ItemName.drill.getItemStack());
-			OreUtil.registerOre("drillDiamond", ItemName.diamond_drill.getItemStack());
-			OreUtil.registerOre("drillAdvanced", ItemName.iridium_drill.getItemStack());
-			
-			OreUtil.registerOre("reflectorBasic", ItemName.neutron_reflector.getItemStack());
-			OreUtil.registerOre("reflectorThick", ItemName.thick_neutron_reflector.getItemStack());
-			OreUtil.registerOre("reflectorIridium", ItemName.iridium_reflector.getItemStack());
+			if(experimental) {
+				OreUtil.registerOre("reflectorIridium", getItem("iridium_reflector"));
+			} else {
+				// Try to get the iridium reflector from the item registry instead.
+				Item item = GameRegistry.findRegistry(Item.class).getValue(new ResourceLocation("ic2:itemreflectors"));
+				if(item == null) {
+					Core.logHelper.warn("Failed to look up the IC2 Classic iridium neutron reflector item (ic2:itemreflectors:2)");
+				} else {
+					System.out.print("Register reflectorIridium with "+new ItemStack(item, 1, 2)+" "+item.getRegistryName());
+					OreUtil.registerOre("reflectorIridium", new ItemStack(item, 1, 2));
+				}
+			}
+
+			ItemStack industrialTnt = getItem("te", "itnt");
+
+			// IC2E ITNT has meta 0 as nukes and meta 1 as industrial tnt. IC2C flips it for some reason.
```

Comment:
Could this be done better? as I can see this breaking and it going noticed for a while.

Not sure of a better way of doing it however.

Reply:
I assume that this comment is in reference to the method I use of retrieving IC2 Classic’s Iridium Reflector item. The root of the problem is that while IC2C implements most of the IC2E api, there are some holes. Since I want to allow the IC2C codepath to compile with the IC2E environment, I decided to retrieve it from the item registry. 

The only other way would be direct access through IC2Items, but that might not be present in IC2E. Perhaps I could fix that issue in the same way that treetaps could be fixed.

However, it doesn’t look like the iridiumReflector oredict is actually used anywhere. The Fusion Coil recipe requires a TR iridium reflector only.

The only way this would fail is if Spieger changes the registry ID, which certainly won’t happen on 1.12z

Also, I just realized that the System.out log message needs to be removed.

Reply:
Ok, that is fair enough. Most of this should be fine during 1.12. 1.13 we can tackle if/when we come it.

Reply:
Yeah, and on that note IC2C probably won’t make it to 1.13+ in the first place... I’ll look into other methods, since I do agree that the method is pretty ugly.